### PR TITLE
Reduce code duplication in audio collection + some small fixes

### PR DESCRIPTION
**Pull request by racoiaws**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Reduce duplicated code across the 6 AudioToAudioModel subclasses by moving shared logic into the base class.                                                     
                                                            
 **Collection**: Audio  

# Changelog

  - Move shared training loop logic (batch parsing, training_step, normalization, init boilerplate) into AudioToAudioModel base class
  - Fix SB component loss logging firing during evaluation                                                                                                         
  - Fix redundant world_size init in EncMaskDec and BNR2       

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

File: nemo/collections/audio/models/audio_to_audio.py [hunk 1]
```diff
@@ -130,6 +133,55 @@
...
+
+        Args:
+            input_signal: input audio tensor (B, C, T)
+            target_signal: target audio tensor (B, C, T)
+            input_length: length of each example in the batch (B,)
+
+        Returns:
+            Scalar loss tensor.
+        """
+        ...
```
[the remark this answers is not in the document]
Comment (racoiaws) [reply nemo/collections/audio/models/audio_to_audio.py]: ... -> pass

File: nemo/collections/audio/models/audio_to_audio.py [hunk 2]
```diff
@@ -313,6 +365,24 @@ def _setup_process_dataloader(self, config: Dict) -> 'torch.utils.data.DataLoade
...
+
+        Args:
+            signal: tensor with shape (B, C, T)
+
+        Returns:
+            Tuple of (normalized_signal, norm_scale). Pass norm_scale to
+            _denormalize to restore the original scale.
+        """
+        norm_scale = torch.amax(signal.abs(), dim=(-1, -2), keepdim=True)
+        return signal / (norm_scale + self.eps), norm_scale
```
Comment (pzelasko) [inline nemo/collections/audio/models/audio_to_audio.py:379]: Should the returned norm_scale include `+ self.eps` for identical reverse operation?

Comment (racoiaws) [reply nemo/collections/audio/models/audio_to_audio.py:379]: Good catch

I'd prefer an explicit add in the reverse operation instead of baking it into `norm_scale`

**Review by anteju — COMMENTED** [review]
LGTM

**Review by anteju — APPROVED** [review]

**Comment by racoiaws** [comment]
/ok to test https://github.com/NVIDIA-NeMo/NeMo/commit/aa51c998c80094ef523114d74cebb1205a607f51
